### PR TITLE
Add staff activity overview

### PR DESCRIPTION
**Pull request by Ewoutvans**
Real simple:

Click tag on profile page of the user
schermafbeelding 2018-01-20 om 22 36 18
schermafbeelding 2018-01-20 om 22 36 22

See what user did, ordered by date:
schermafbeelding 2018-01-20 om 22 36 26

### CODE REVIEW

File: app/views/users/admin/activity.scala.html [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+            <div class="col-md-12">
+                <table class="table table-condensed setting-no-border table-review-log">
+                    <tbody>
+                        @if(activities.isEmpty) {
+                            <tr><td>No activities founds</td></tr>
+                        }
+                        @for(activity <- activities) {
+                            @if(activity.isInstanceOf[Review]) {
+                                <tr>
+                                    <td>Review approved:</td>
```
Comment (phase) [inline app/views/users/admin/activity.scala.html]: Have this link to the review

File: app/views/users/admin/activity.scala.html [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+                                    <td>@prettifyDateAndTime(activity.asInstanceOf[Review].endedAt.getOrElse(Timestamp.from(Instant.EPOCH)))</td>
+                                </tr>
+                            }
+                            @if(activity.isInstanceOf[Flag]) {
+                                <tr>
+                                    <td>Flag resolved:</td>
```
Comment (phase) [inline app/views/users/admin/activity.scala.html]: Have this link to whatever project got flagged (or if there's a "flag" page link to that)

File: app/views/users/view.scala.html [hunk 3]
```diff
@@ -141,7 +142,13 @@ <h1 class="username">
 
                     @defining(user.globalRoles.toList.sortBy(_.trust).lastOption) { roleOpt =>
                         @roleOpt.map { role =>
-                            <li class="user-role channel" style="background-color: @role.color.hex">@role.title</li>
+                            @if(users.current.isDefined && users.current.get.can(ViewActivity).in(user)) {
```
Comment (phase) [inline app/views/users/view.scala.html]: `users.current.get can ViewActivity in user ` would be more consistent with the rest of the code

File: app/controllers/Application.scala [hunk 4]
```diff
@@ -186,4 +189,30 @@ final class Application @Inject()(data: DataHelper,
...
+    if (u.id.isDefined) {
+      val reviews = this.service.access[Review](classOf[Review]).filter(_.userId === u.id.get)
+      val flags = this.service.access[Flag](classOf[Flag]).filter(_.resolvedBy === u.id.get)
+      objects = reviews ++ flags
+      objects.sortWith(sortMethod)
+    }
+    Ok(views.users.admin.activity(u, objects))
+  }
+
+  def sortMethod(o1: Object, o2: Object): Boolean = {
```
Comment (phase) [inline app/controllers/Application.scala]: Can we get a better name for this? `compareObjects` or something.

**Review by phase — CHANGES_REQUESTED** [review]

**Comment by phase** [comment]
Why is the link on the tag? I think a calendar icon next to the existing buttons would look more consistent.

The activity page is also very bland. Maybe a white background underneath the text?

**Comment by Ewoutvans** [comment]
@phase I changed it to a calendar & added the table to a panel so its more like the rest of the site
I've updated the screenshots

**Review by kashike — APPROVED** [review]

**Review by phase — APPROVED** [review]

**Pull request merged**